extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalogAdapter.java

```diff
@@ -860,8 +835,456 @@ private Table readTableInternal(
+            @NotNull final Table dhTable,
+            @Nullable final IcebergWriteInstructions instructions) {
+        return writeDataFiles(tableIdentifier, new Table[] {dhTable}, instructions);
+    }
+
+    /**
+     * Writes data from Deephaven tables to an Iceberg table without creating a new snapshot. This method returns a list
+     * of data files that were written. Users can use this list to create a transaction/snapshot if needed. All tables
+     * should have the same definition, else a table definition should be provided in the instructions.
+     *
+     * @param tableIdentifier The identifier string for the Iceberg table to write to.
+     * @param dhTables The Deephaven tables containing the data to be written.
+     * @param instructions The instructions for customizations while writing, or null to use default instructions.
+     *
+     * @return A list of {@link DataFile} objects representing the written data files.
+     */
+    public List<DataFile> writeDataFiles(
+            @NotNull final String tableIdentifier,
+            @NotNull final Table[] dhTables,
+            @Nullable final IcebergWriteInstructions instructions) {
+        return writeDataFiles(TableIdentifier.parse(tableIdentifier), dhTables, instructions);
+    }
+
+    /**
+     * Writes data from Deephaven tables to an Iceberg table without creating a new snapshot. This method returns a list
+     * of data files that were written. Users can use this list to create a transaction/snapshot if needed. All tables
+     * should have the same definition, else a table definition should be provided in the instructions.
+     *
+     * @param tableIdentifier The identifier for the Iceberg table to write to.
+     * @param dhTables The Deephaven tables containing the data to be written.
+     * @param instructions The instructions for customizations while writing, or null to use default instructions.
+     *
+     * @return A list of {@link DataFile} objects representing the written data files.
+     */
+    public List<DataFile> writeDataFiles(
+            @NotNull final TableIdentifier tableIdentifier,
+            @NotNull final Table[] dhTables,
+            @Nullable final IcebergWriteInstructions instructions) {
+        return writeImpl(tableIdentifier, dhTables, instructions, false, false);
+    }
+
+    /**
+     * Appends or overwrites data in an Iceberg table with the provided Deephaven tables.
+     *
+     * @param tableIdentifier The identifier for the Iceberg table to append to or overwrite
+     * @param dhTables The Deephaven tables to write
+     * @param instructions The instructions for customizations while writing, or null to use default instructions
+     * @param overwrite If true, the existing data in the Iceberg table will be overwritten; if false, the data will be
+     *        appended
+     * @param addSnapshot If true, a new snapshot will be created in the Iceberg table with the written data
```

Comment:
`overwrite` requires `addsnapshot`. I know it's a private method, but we should validate that. No need to make the interface any prettier than that, since this is internal.